src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,229 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.inner_product_space.adjoint
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `P _ᗮ` for `orthogonal_projection _ᗮ`.
+
+We let `V` be a finite-dimensional inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+local notation `P`U := (orthogonal_projection U)
+local notation `P`U`ᗮ` := (orthogonal_projection Uᗮ)
+
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def invariant_subspace (U : submodule ℂ V) (T : V →L[ℂ] V) :
+ Prop := U ≤ U.comap T
+lemma invariant_subspace_def (U : submodule ℂ V) (T : V →L[ℂ] V) :
+ invariant_subspace U T ↔ (T '' U ⊆ U)
+  := by simp only [set.image_subset_iff]; refl
+
+/-- `U` is `T` invariant if and only if `(P U) * T * (P U) = T * (P U)` -/
+lemma subspace_is_invariant_iff_ortho_proj_mul_T_mul_ortho_proj_eq_T_mul_ortho_proj
+  [finite_dimensional ℂ V] (U : submodule ℂ V) (T : V →L[ℂ] V) :
+  (invariant_subspace U T) ↔ ∀ x : V, ↑((P U) (T ↑((P U) x))) = (T ↑((P U) x)) :=
+begin
+  split,
+  { intros h x,
+    obtain ⟨w, hw, v, hv, hvw⟩ := submodule.exists_sum_mem_mem_orthogonal U x,
+    rw [hvw, map_add,
+      orthogonal_projection_mem_subspace_orthogonal_complement_eq_zero hv,
+      add_zero, orthogonal_projection_eq_self_iff.mpr hw],
+    exact orthogonal_projection_eq_self_iff.mpr (h hw) },
+  { intros h u h_1,
+    rw [submodule.mem_comap,
+      ← orthogonal_projection_eq_self_iff, ← orthogonal_projection_eq_self_iff.mpr h_1, h] }
+end
+
+/-- `U,Uᗮ` are `T` invariant if and only if `(P U) * T = T * (P U)` -/
+lemma U_and_U_bot_are_T_invariant_iff_ortho_proj_mul_T_eq_T_mul_ortho_proj
```

Comment:
```suggestion
lemma U_and_U_orthogonal_are_T_invariant_iff_ortho_proj_mul_T_eq_T_mul_ortho_proj
```
`ᗮ` is `orthogonal` not `bot`. Same throughout

Reply:
I changed them all to `ortho` now